### ISSUE TITLE
Refrigerants R134a and R410a for modular heat pumps

### ISSUE THREAD

**Issue by MirkoEngelpracht** [RWTH-EBC/AixLib#408]
## What is the problem?

This issue is a work package for issue #407.
Currently, there are no refrigerants within the AixLib library. Therefore, we want to add two different refrigerants (R134a and probably R410a).

## Why do we want to solve it?

We want to solve this for issue #407: The refrigerants can be used, for example, for heat pumps and refrigerating machines.

## How do we want to solve it?

We will use the approach presented by Sangi et al. (see paper “A Medium Model for the Refrigerant Propane for Fast and Accurate Dynamic Simulations”; [http://www.ebc.eonerc.rwth-aachen.de/go/id/dncb/file/227214](url)). Sangi et al. calculated functions for the density, specific enthalpy and specific entropy for the three different regimes supercooled, superheated and two-phase. These functions were polynomial and fitted to the inverse energy equation of state.

## Who do we think will benefit from our solution?

We think that everyone will benefit from the models who needs refrigerants without using external media libraries. Moreover, the presented approach allows a fast and accurate simulation and, therefore, contributes to everyone who will focus the simulation speed.

**Comment by MirkoEngelpracht** [comment]
Hello everybody,
the issue is solved and ready to be reviewed.

I've added a few examples and validation models. Within the validation models, I've checked all medium functions including the fitted formulas as well as the derivatives. Furthermore, I've compared the results to external media libraries, i.e. the HelmholtzMedia library and ExternalMedia library.

Actually, some problems occur while using partial derivatives with respect to specific entropy. Moreover, I'm not quite satisfied with my solution of implementing a general form of two-dimensional polynomials as done, for example, for the function 'density_pT'. I've also added the coefficients of fitted formulas directly in the model for reasons of simulation speed.

Perhaps somebody has some ideas to provide a nicer solution for the issues mentioned above.

Best regards
Mirko
